### PR TITLE
[dti48] Allow user to choose a different semester of data

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

Fetch all semesters' data and let user choose. Step 2 of making DTI 48 playable for old members. Next step would be allow old members to log in.

### Test Plan 

Screen Shot 2021-08-30 at 15 34 02

### CODE REVIEW

File: frontend/src/Games/DTI48/DTI48.tsx [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+}
+
+async function getMembersFromAllSemesters(): Promise<MembersFromAllSemesters> {
+  const [currentSemesterMembers, sp21Members] = await Promise.all([
+    getMembers(),
+    getMembers('sp21')
+  ]);
+  return {
+    'Current Semester': currentSemesterMembers,
+    'Spring 2021': sp21Members
```
Comment (henry-li-06) [inline frontend/src/Games/DTI48/DTI48.tsx:21]: Is there a way to not hardcode `sp21Members`, `'sp21'`, and `'Spring 2021'`

**Review by henry-li-06 — COMMENTED** [review]
In `computeDTI49UpgradeChain` what happens to leads from previous semesters when the list of ids for each of the role leads changes. E.g. Jagger is former dev lead so his role should be a `lead` but `DEV_LEAD_NETIDS` for this semester should be Scott and Enoch.

**Comment by SamChou19815** [comment]
> In `computeDTI49UpgradeChain` what happens to leads from previous semesters when the list of ids for each of the role leads changes. E.g. Jagger is former dev lead so his role should be a `lead` but `DEV_LEAD_NETIDS` for this semester should be Scott and Enoch.

Yes, I do plan to address that in another PR, potentially by making each lead have a subteam that can identify their types (e.g. devleads), then we can get rid of this hardcoding and make the tree generation fully automated.

File: frontend/src/Games/DTI48/DTI48.tsx (hunk 1 above)
Comment (SamChou19815) [reply frontend/src/Games/DTI48/DTI48.tsx:21]: There is probably a way to make the backend return an API like this, but I plan to address it in another PR where I add the login support for achieved members.

**Review by henry-li-06 — APPROVED** [review]

**Pull request merged**